### PR TITLE
Makefile targe to disable certification check

### PR DESCRIPTION
**Pull request by tcrain**
The idea here is to be able to disable the certification check at compile time by using the make target:
make relnocert

This way we don't have to make separate branches for benchmarking.  I guess eventually the cert check will be done dynamically deeding on the object and what the client wants.
The issues with the certification check are described here:
https://github.com/SyncFree/antidote/issues/131
https://github.com/SyncFree/antidote/issues/130

### CODE REVIEW

File: rebar.config.script [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+case os:getenv("NO_CERTIFICATION") of
+    false ->
+        CONFIG;
+    _ ->
+        case lists:keysearch(erl_opts, 1, CONFIG) of
+            {value, {erl_opts, Opts}} ->
+                NewOpts = [NoCert | Opts],
+                lists:keyreplace(erl_opts, 1, CONFIG, {erl_opts, NewOpts});
+            false ->
+                CONFIG ++ [{erl_opts, NoCert}]
```
Comment (bieniusa) [inline rebar.config.script]: Is it necessary to append to the end? It is in general better to append at the front.

**Comment by bieniusa** [comment]
Tyler, can you resolve the conflicts with master?

I would at least add another Makefile target to run the test suite under different configurations (i.e. with and without the check). And one large target that runs all the configurations.

**Comment by tcrain** [comment]
If you run
make rel
It is the same as before (it includes the certification check).

If you run
make relnocert
Then when you run the tests it will only run the tests that don't require the certification check.

They both get built to the same "rel" directory, so you can only have one version at a time.
Otherwise I could make them build to different directories and configure the tests to run in both directories. Is this what you mean?

I guess the goal is to eventually be able to build antidote with different consistencies so maybe we should have a standard way of doing this.

Otherwise the rebar profiles seem like something interesting to use here, but needs rebar3
https://www.rebar3.org/docs/profiles

**Comment by bieniusa** [comment]
> Am 29.09.2015 um 20:07 schrieb tcrain <notifications@github.com>:
> 
> If you run
> make rel
> It is the same as before (it includes the certification check).
> 
> If you run
> make relnocert
> Then when you run the tests it will only run the tests that don't require the certification check.
> 
> They both get built to the same "rel" directory, so you can only have one version at a time.
> Otherwise I could make them build to different directories and configure the tests to run in both directories. Is this what you mean?

Yes, having one make target that builds and tests both configuration options would be necessary at least for use in the CI server.

**Comment by bieniusa** [comment]
@tcrain Please fix conflicts

**Comment by tcrain** [comment]
Conflicts are resolved.

Also I created an issue https://github.com/SyncFree/antidote/issues/165 to track this until we come up with a more complete way to build and test different versions.

**Pull request merged**